authority/authority.go

```diff
@@ -601,6 +595,12 @@ func (a *Authority) IsRevoked(sn string) (bool, error) {
 	return a.db.IsRevoked(sn)
 }
 
+// GetIntermediateCertificate returns the x509 intermediate CA
+// certificate.
+func (a *Authority) GetIntermediateCertificate() (*x509.Certificate, error) {
+	return pemutil.ReadCertificate(a.config.IntermediateCert)
+}
```

Comment:
Not all the configurations of step-ca will have an intermediate file. For example, by default, an RA will not, and SCEP is a great candidate for an RA, although it might not be supported yet because it requires an RSA certificate and key for decrypt/encrypt things.

To be future proof, we should get the intermediate from `authority.x509CAService`, but right now it does not have a way to do it directly. The other option is to get it from the step-ca own certificate, generated in `authority.GetTLSCertificate()`.

Reply:
There are indeed improvements to how SCEP can be configured and used. I think that should be combined in a bigger package of work related to SCEP after discussing what types of use cases to support and how.